src/core_plugins/elasticsearch/lib/set_headers.js

```diff
@@ -0,0 +1,15 @@
+import { isObject } from 'lodash';
+
+export default function setHeaders(originalHeaders, newHeaders) {
+  if (!isObject(originalHeaders)) {
```

Comment:
I considered that, but spread should work just fine with any enumerable object, so I couldn't think of a reason to only allow plain objects.